packages/core/echo/echo-db/src/query/query-state.ts

```diff
@@ -162,8 +162,13 @@ export class QueryState extends Resource {
 
 // TODO(burdon): Process Filter DSL.
 const filterToIndexQuery = (filter: Filter): IndexQuery => {
-  return {
-    typename: filter.type?.itemId,
-    inverted: filter.not,
-  };
+  if (filter.type || (filter.or.length > 0 && filter.or.every((subFilter) => !subFilter.not && subFilter.type))) {
```

Comment:
is it guaranteed that filter.or and filter.type are exclusive?

Reply:
Good question, but how would you resolve filter with `type` and `or` section